### PR TITLE
Include macOS sandbox files in the Nix binary

### PR DESCRIPTION
**Pull request by edolstra**
This basically reverts https://github.com/NixOS/nix/commit/6e5165b77370c76bfa39d4b55e9f83673f3bd466. It fixes errors like

    sandbox-exec: <internal init prelude>:292:47: unable to open sandbox-minimal.sb: not found

when trying to run a development Nix installed in a user's home directory.

Also, we're trying to minimize the number of installed files to make it possible to deploy Nix as a single statically-linked
binary.

### CODE REVIEW

File: src/libmain/shared.cc [hunk 1]
```diff
@@ -363,6 +363,7 @@ void printVersion(const std::string & programName)
             << "\n";
         std::cout << "Store directory: " << settings.nixStore << "\n";
         std::cout << "State directory: " << settings.nixStateDir << "\n";
+        std::cout << "Data directory: " << settings.nixDataDir << "\n";
```
Comment (thufschmitt) [inline src/libmain/shared.cc:366]: Is that related to the rest?

**Review by thufschmitt — APPROVED** [review]
Just one weird apparently unrelated change, but looks good otherwise :+1:

File: src/libmain/shared.cc (hunk 1 above)
Comment (edolstra) [reply src/libmain/shared.cc:366]: Not really related, it was just useful during debugging this issue.
